src/main/java/com/example/demo/ProductController.java

```diff
@@ -0,0 +1,37 @@
+package com.example.demo;
+
+import com.example.demo.domain.ProductFacade;
+import com.example.demo.domain.ProductRequestDto;
+import com.example.demo.domain.ProductResponseDto;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/products")
+class ProductController {
+
+    private final ProductFacade productFacade;
+
+    public ProductController(ProductFacade productFacade) {
+        this.productFacade = productFacade;
+    }
+
+    @PostMapping()
+    ProductResponseDto createProduct(@RequestBody ProductRequestDto rd) {
+        return productFacade.create(rd);
+    }
+
+    @GetMapping("/{id}")
+    ProductResponseDto getProduct(@PathVariable("id") String id) {
+        return productFacade.find(id);
+    }
+
+    @PatchMapping("/{id}")
+    ProductResponseDto updateProduct(@PathVariable(value = "id") String id, @RequestParam("name") String name) {
```

Comment:
Update produktu, powinien się odbywać z użyciem metody `PUT`, której przekażemy DTO ze wszystkimi polami (nawet jeśli wartości się nie zmieniają). Taki jest standard w RESTowym podejściu. Jeśli chciałbyś użyć metody `PATCH` to jest taka opcja, ale nie jest to tak proste jak się wydaje :)

W rozwiązaniu, które wybrałeś widzę dwa problemy:
1) Gdybyś chciał aktualizować kilka pól to musiałbyś tutaj dodawać dodatkowe argumenty do funkcji, dlatego lepiej stworzyć osobne DTO, które zbierze wszystkie argumenty.

2) Wg. RFC dla metody `PATCH` dane, które przesyłasz powinny być listą zmian, wskazującą co i gdzie powinno się zmienić. Powinieneś zatem dla każdej takiej zmiany wskazać ścieżkę oraz nową wartość. Przykładowe body:

```
{
  changes: [
    { path: "name", value: "new product name" }
  ]
}
```

Wtedy po stronie aplikacji pozostaje przeiterowanie się po tych zmianach i zaaplikowanie zmian wskazanych przez `path` pól na nowe wartości wskazane w `value`. 

Zazwyczaj jest to gra niewarta świeczki i lepiej użyć `PUT` i wysłać po prostu cały produkt jeszcze raz z nowymi wartościami.